File: output/qwen_2.5_72b_instruct/codenet/algo_based_translation/python/java/s259116349.java
import java.util.ArrayList;
import java.util.List;
import java.util.Scanner;

public class s259116349 {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        int N = scanner.nextInt();
        int[] primes = {2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47};
        List<int[]> divs = new ArrayList<>();

        for (int i = 1; i <= 100; i++) {
            int[] tmp = new int[primes.length];
            for (int j = 0; j < primes.length; j++) {
                int p = primes[j];
                tmp[j] = 0;
                while (i / p == (double) i / p) {
                    i = i / p;
                    tmp[j]++;
                }
            }
            divs.add(tmp);
        }

        int[] divsum = new int[primes.length];
        for (int i = 1; i <= N; i++) {
            for (int j = 0; j < primes.length; j++) {
                divsum[j] += divs.get(i - 1)[j];
            }
        }

        int ans = 0;
        ans += num(divsum, 75);
        ans += num(divsum, 15) * (num(divsum, 5) - 1);
        ans += num(divsum, 25) * (num(divsum, 3) - 1);
        ans += num(divsum, 5) * (num(divsum, 5) - 1) * (num(divsum, 3) - 2) / 2;
        System.out.println(ans);
    }

    public static int num(int[] divsum, int n) {
        int count = 0;
        for (int value : divsum) {
            if (value >= n - 1) {
                count++;
            }
        }
        return count;
    }
}

File: dataset/codenet/Python/Code/s259116349.py
from math import factorial as f

N = int(input())

primes = [2,3,5,7,11,13,17,19,23,29,31,37,41,43,47]
divs = []
for i in range(1,101):
    tmp = [0 for _ in range(len(primes))]
    for j in range(len(primes)):
        p = primes[j]
        tmp[j] = 0
        while(i//p==i/p):
            i = i//p
            tmp[j] += 1
    divs.append(tmp)
#print(divs)

divsum = [0 for _ in range(len(primes))]
for i in range(1, N+1):
    for j in range(len(primes)):
        divsum[j] += divs[i-1][j]
#print(divsum)

def num(n):
    return len(list(filter(lambda x:x>=n-1, divsum)))

ans = 0

# >=74
ans += num(75)

# >=14 & 13~4
ans += num(15)*(num(5)-1)

# >=24 & 23~2
ans += num(25)*(num(3)-1)

# >=4,>=4 & >=2
ans += num(5) * (num(5)-1) * (num(3)-2) // 2 
print(int(ans))
